### ISSUE TITLE
[BUG]: In some situationes textures won't be replaced (Texture replacement feature)

### ISSUE THREAD

**Issue by Kupo91** [PCSX2/pcsx2#9968]
### Describe the Bug

In a few situations pcsx2 does not replace textures, even if it dumped them properly.
Best visible in this room:

![Dragon Quest VIII - Journey of the Cursed King_SLES-53974_20230916131718](https://github.com/PCSX2/pcsx2/assets/41426906/73f3e8cf-ba04-4696-9487-51021ea76db0)

The textures which have not been replaced have short file names.

![Unbenannt](https://github.com/PCSX2/pcsx2/assets/41426906/ebaacf40-4986-4f8a-bb05-d27c4fbdfe41)

**Tested in v1.7.5013**

### Reproduction Steps

Savegame and savestate are attached: 
[save_files.zip](https://github.com/PCSX2/pcsx2/files/12640541/save_files.zip)

If you want to you use the savegame, go to the location this way:

- Leave the house
- Use zoom to teleport to marta's cottage
- Enter marta's house
- Go to the first room in the basement

Enable texture dumping, edit the texture files which have short file names, enable texture replacing and see what is (not) happening

### Expected Behavior

Also textures with short file names can be replaced

### PCSX2 Revision

1.7.5013

### Operating System

Windows 10 (64bit)

### CPU

i7-4790

### GPU

GT 1030

### Logs & Dumps

Wasn't sure if a GS dump is needed, so I just made one: 
[gs_dump.zip](https://github.com/PCSX2/pcsx2/files/12640542/gs_dump.zip)

**Comment by xXthecatXx** [comment]
![Dragon Quest VIII - Journey of the Cursed King_SLES-53974_20230916165518](https://github.com/PCSX2/pcsx2/assets/127200899/8155575c-fa0e-463b-b2ce-b5fff3ec9c13)
it works fine for me it probably doesn't work for you because you don't delete in .png name

**Comment by Kupo91** [comment]
> it works fine for me it probably doesn't work for you because you don't delete in .png name

hm don't delete what?

**Comment by xXthecatXx** [comment]
![Screenshot_1](https://github.com/PCSX2/pcsx2/assets/127200899/279a5761-06c4-4bc6-af27-d1098c3c235b)
replace it without the .png name

**Comment by Kupo91** [comment]
Ah, no that was not the problem. I encountered this issue already half a year ago. Since there is a new release coming soon, I now gave it another try and as I thought, my HD textures still didn't show up. After you showed me, that it works like intended, I noticed that my old files have slightly different names. Looks like there was an update which changed the first part of the hash in these cases. With the new ones it works indeed. So yeah, sorry, my bad. Should have checked it more carefully. Good to know that it works now.
